tests/distributions/test_hmm.py

```diff
@@ -334,6 +335,24 @@ def test_gaussian_hmm_shape(diag, init_shape, trans_mat_shape, trans_mvn_shape,
         assert d2.event_shape == (f, obs_dim)
 
 
+def test_gaussian_hmm_high_obs_dim():
+    hidden_dim = 1
+    obs_dim = 1000
+    duration = 10
```

Comment:
Those are limits in my system to be able to run the test with the old implementation. With the new implementation in this PR, we don't have OOM issue.